### ISSUE TITLE
Creating a lot of relationships at once causes a timeout error

### ISSUE THREAD

**Issue by Lhorus6** [LuatixHQ/opencti#6361]
## Description

When we launch the creation of a large number of relationships in a knowledge graph, the creation process takes too long and a timeout error occurs. Here's an example of a use case where this happens:
I have 350 IOCs in a report and one malware. I want to link all the IOCs with my malware, but the error occurs.

## Environment

OCTI 6.0.6

## Reproducible Steps

Steps to create the smallest reproducible scenario:
1. Create a repot
2. Go in the entities tab
3. Add one malware
4. Add 400 Indicators
5. Go in the knowledge graph
6. Try to create a relation between all indicators and the malware (in one run)

## Screenshots
![image](https://github.com/OpenCTI-Platform/opencti/assets/70956706/f72fb7fb-d7f1-43e5-9556-fb274c63221f)

**Comment by SamuelHassine** [comment]
I confirm this has been reported several times. This is something we should address in the backend.

Current situation:

* The frontend create 400 relationships 1 by 1

Whished situation:

* The frontend send all relationships to create all things at once.

**Comment by lndrtrbn** [comment]
> The frontend send all relationships to create all things at once.

Does this kind of API already exists and it's just frontend code to change? Or do we also need to create this batch relationships creation API?

**Comment by nino-filigran** [comment]
@SouadHadjiat @aHenryJard @labo-flg Do you have any idea about the above question? ☝️

**Comment by SamuelHassine** [comment]
We need a new API "stixCoreRelationshipsAdd" (plurial) in addition to "stixCoreRelationshipAdd" to batch creation of relationships.

**Comment by richard-julien** [comment]
This one is really complex to achieve. Please come talk to me before starting it

**Comment by SamuelHassine** [comment]
Assigning to Julien since we are working on removing locks on relationships nodes.

**Comment by SamuelHassine** [comment]
Will be resolved by #6361.

**Comment by Lhorus6** [comment]
Hi @SamuelHassine,
Unfortunately, the problem is still present :/ 

## [My test here](https://demo.octi.filigran.io/dashboard/analyses/groupings/3996e48e-6556-4867-a7d2-fa57c9a7f7d1/knowledge/graph?zoom=%7B%22k%22%3A0.1062019962696128%2C%22x%22%3A7843.210378267719%2C%22y%22%3A517.8810373806218%7D&mode3D=false&selectRectangleModeFree=false&selectModeFree=false&selectModeFreeReady=true&modeFixed=false&modeTree=&displayTimeRange=false&selectedTimeRangeInterval=Tue+Oct+29+2024+00%3A00%3A00+GMT%2B0100+%28Central+European+Standard+Time%29%2CWed+Oct+30+2024+23%3A59%3A59+GMT%2B0100+%28Central+European+Standard+Time%29&stixCoreObjectsTypes=&markedBy=&createdBy=&width=null&height=null&zoomed=false&keyword=&openCreatedRelation=false):

![image](https://github.com/user-attachments/assets/3c93f1db-c014-4ecb-a57a-2e6c853ea3e5)

## The logs:

`{
    "category": "APP",
    "errors": [
        {
            "attributes": {
                "genre": "BUSINESS",
                "http_status": 500,
                "participantIds": [
                    "3996e48e-6556-4867-a7d2-fa57c9a7f7d1",
                    "grouping--b49dd1ed-f6a4-5b4d-bf7c-e99f48c5bdc8"
                ]
            },
            "message": "Execution timeout, too many concurrent call on the same entities",
            "name": "LOCK_ERROR",
            "stack": "GraphQLError: Execution timeout, too many concurrent call on the same entities\n    at error (/opt/opencti/build/src/config/errors.js:7:10)\n    at LockTimeoutError (/opt/opencti/build/src/config/errors.js:109:51)\n    at updateAttributeMetaResolved (/opt/opencti/build/src/database/middleware.js:2188:13)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at patchElementWithRefRelationships (/opt/opencti/build/src/domain/stixObjectOrStixRelationship.ts:42:36)\n    at stixObjectOrRelationshipAddRefRelation (/opt/opencti/build/src/domain/stixObjectOrStixRelationship.ts:60:23)"
        }
    ],
    "inner_relation_creation": 0,
    "level": "error",
    "message": "Execution timeout, too many concurrent call on the same entities",
    "operation": "Unspecified",
    "query_attributes": [
        [
            {
                "arguments": [
                    [
                        {
                            "is_empty": true,
                            "name": "id",
                            "type": "Variable"
                        }
                    ],
                    [
                        {
                            "is_empty": true,
                            "name": "input",
                            "type": "Variable"
                        }
                    ]
                ],
                "name": "groupingRelationAdd"
            }
        ]
    ],
    "size": 130,
    "source": "backend",
    "time": 35455,
    "timestamp": "2024-10-30T20:09:52.042Z",
    "type": "WRITE_ERROR",
    "user": {
        "group_ids": [
            "77e6b343-5fc4-4f5b-b760-6c889f32bc68",
            "642f892a-ad4f-4460-a989-b19a0a1c95cd"
        ],
        "ip": "79.85.59.193",
        "organization_ids": [
            "21e1c0ce-c28d-498e-aaf9-211680ebc6a2"
        ],
        "referer": "https://demo.octi.filigran.io/dashboard/analyses/groupings/3996e48e-6556-4867-a7d2-fa57c9a7f7d1/knowledge/graph?zoom=%7B%22k%22%3A0.10620199626961285%2C%22x%22%3A1016.8284396202927%2C%22y%22%3A245.10971216366067%7D&mode3D=false&selectRectangleModeFree=true&selectModeFree=false&selectModeFreeReady=true&modeFixed=false&modeTree=&displayTimeRange=false&selectedTimeRangeInterval=Tue+Oct+29+2024+00%3A00%3A00+GMT%2B0100+%28Central+European+Standard+Time%29%2CWed+Oct+30+2024+23%3A59%3A59+GMT%2B0100+%28Central+European+Standard+Time%29&stixCoreObjectsTypes=&markedBy=&createdBy=&width=null&height=null&zoomed=false&keyword=&openCreatedRelation=false",
        "socket": "query",
        "user_id": "bc2562f1-293b-4cdf-841a-f834b27fdbb0",
        "user_metadata": {}
    },
    "version": "6.3.8"
}`

## Next step

Do you want me to open a new issue ? Or do we reopen that one ?

**Comment by nino-filigran** [comment]
@Lhorus6 you tried to create a relation from what to what? (a rough amount of relation would be useful)

**Comment by Lhorus6** [comment]
Hi @nino-filigran,
You can find the link to my test on the text "My test here" in my previous message. 

I launched the creation of 798 relations "Indicator -> indicates -> Incident". But it does not matter, try to make several hundred relations at once and you will have the same problem no matter the entity type.

**Comment by SamuelHassine** [comment]
Definitely resolved in the upcoming minor release (+ progress bar).
